book/src/out-of-order-blocks.md

```diff
@@ -0,0 +1,130 @@
+
+* *N* number of bits of a leader's vote signature that must match a PoH hash.
+This hash can be in between ticks.
+
+The leader must find the golden ticket in the empty PoH entries leading up to
+the proposed block.  Once the golden ticket is found, the leader can propose the
+block for the next slot.
+
+To find the golden ticket, the leader signs the BlockHash of the block that the
+entries are generated from with a Vote.  Since each validator is resetting its
+PoH every time it votes, each validator can search for a golden ticket from that
+starting point, and look for a PoH hash matching *N* bits of the Vote signature.
+
+The same proposed [fork selection]/fork-selection.md) rules apply.  Once a block
+is voted on, the validator would switch its PoH to reset on this new block, and
+either a new golden ticket block is proposed, or the next scheduled leader
+proposes a block.
+
+To make it easy to verify, the vote with the golden ticket must be present in
+the proposed block as the first transaction.
+
+### Grinding
+
+Because the golden ticket is biasable, the last leader has the most opportunity
+to influence the BlockHash to generate the golden ticket for itself.  This
+BlockHash could be a grinding attack with N cores.  Grinding should be a limited
+attack vector, since the difficulty is determined by the size of the delegates
+to the leaders’ VoteState account.
+
+To reduce the ability to grind the value, the golden ticket must appear *X*
+ticks after the block, where *X* is half of the number of ticks per block.
+
+### Difficulty
+
+Difficulty is the number of matching bits that the golden ticket needs to match
+the signature.
+
+The golden ticket connects via PoH entries to a known block with an immutable
+account state.  The stake size can be used to set the difficulty.  Hashing with
+1 account has been delegated 100 tokens should generate as many tickets as 2
+accounts with 50 delegated tokens.
+
+Given the total delegated stake size for the epoch, and number of hashes per
+Tick, the difficulty can be computed with the leader scheduler.
+
+### Generating VDFs for Multiple Forks
+
+The number of potential forks the attacker can choose from is limited by
+the fork selection algorithm.  The generated golden ticket needs to be past the
+lockout distance for the rest of the cluster and also be the heaviest fork
+based on lockout.  Let’s say the difficulty was set to the cluster generating a
+golden ticket block every 3 slots.
+
+For example:
+
+Starting with ‘n - 3’ forks, the proposed block would only be valid if no one in
+the cluster voted.  If they did, the golden ticket would need to be in the ‘n +
+7’ slot and propose the n + 8 block.
+
+### Transaction Ingress Point
+
+Because there is no deterministic ingress point, there is no place for
+validators to forward transactions.  This design will end up requiring a shared
+*mempool* of transactions between all the nodes, just like traditional PoW
```

Comment:
Does that mean forwarding is no longer required?

Reply:
still required :).  let me remove this